assignments/arrays.js

```diff
@@ -63,34 +63,122 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+
+// First, I describe a perfectly generalist implementation to retrieve an array which has the values for a given key, array of values for a given key. Originally it was nested but I extracted it so I could fiddle with it in other things if needed.
+function arrayListValueByKey(array,key) {
+  let returnArray = []
+  for (i = 0; i < array.length; i++) {
+    returnArray.push(array[i][key]);
+  };
+  return returnArray;
+}
+
+function describeCarByID(IDnum){
+  let returnArray = arrayListValueByKey(inventory,"id");
+  // this is just an error log in case something breaks in my code.
+  if (returnArray[0] === undefined) { console.log('Key array not extractable!')}
+  // confirmation log
+  else if (returnArray[0] !== false) { console.log('Key array extractable!'); };
+  // I now invoke the result...
+  // Now I invoke an indexOf on our use case and save it!
+  let carIndex = returnArray.indexOf(IDnum);
+  // Now I return a string describing the car at that index, so it can be console logged.
+  return 'Car ' + IDnum + ' is a ' + inventory[carIndex]["car_year"] + ' ' + inventory[carIndex]["car_make"] + ' ' + inventory[carIndex]["car_model"] + '.';
+};
+
+console.log(describeCarByID(33));
 
 
 
 // ==== Challenge 2 ====
```

Comment:
good job. even if the database changes, you will still always pull the last one.